frontend/src/pages/notebookController/SpawnerPage.tsx

```diff
@@ -23,69 +23,91 @@ import {
+const SpawnerPage: React.FC = React.memo(() => {
   const history = useHistory();
-  const { images, dashboardConfig } = React.useContext(NotebookControllerContext);
-  const [username, namespace] = useSelector<State, [string, string]>((state) => [
-    state.appState.user || '',
-    state.appState.namespace || '',
-  ]);
+  const notification = useNotification();
+  const { images, loaded, loadError } = useWatchImages();
+  const { currentUserState, setCurrentUserState } = React.useContext(NotebookControllerContext);
+  const stateUsername = useSelector<State, string>((state) => state.appState.user || '');
+  const username = currentUserState.user || stateUsername;
+  const namespace = useSelector<State, string>((state) => state.appState.namespace || '');
   const projectName = ODH_NOTEBOOK_REPO || namespace;
-  const translatedUsername = usernameTranslate(username);
-  const { buildStatuses } = React.useContext(AppContext);
+  const [startShown, setStartShown] = React.useState<boolean>(false);
+  const { notebook, notebookLoaded } = useWatchNotebookForSpawnerPage(
+    startShown,
+    projectName,
+    username,
+  );
+  const isNotebookRunning = checkNotebookRunning(notebook);
+  const { buildStatuses, dashboardConfig, setIsNavOpen } = React.useContext(AppContext);
   const [selectedImageTag, setSelectedImageTag] = React.useState<ImageTag>({
     image: undefined,
     tag: undefined,
   });
   const [sizeDropdownOpen, setSizeDropdownOpen] = React.useState<boolean>(false);
   const [selectedSize, setSelectedSize] = React.useState<string>('');
-  const [gpuDropdownOpen, setGpuDropdownOpen] = React.useState(false);
+  const [gpuDropdownOpen, setGpuDropdownOpen] = React.useState<boolean>(false);
   const [selectedGpu, setSelectedGpu] = React.useState<string>('0');
   const [gpuSize, setGpuSize] = React.useState<number>(0);
   const [variableRows, setVariableRows] = React.useState<VariableRow[]>([]);
   const [createInProgress, setCreateInProgress] = React.useState<boolean>(false);
-  const userState = React.useMemo(() => {
-    if (translatedUsername) {
-      const newUserState = dashboardConfig?.status?.notebookControllerState?.find(
-        (state) => state.user === translatedUsername,
+  const [shouldRedirect, setShouldRedirect] = React.useState<boolean>(true);
+
+  const onModalClose = () => {
+    setStartShown(false);
+    if (notebook) {
+      deleteNotebook(projectName, notebook.metadata.name).catch((e) =>
+        notification.error(`Error delete notebook ${notebook.metadata.name}`, e.message),
       );
-      if (newUserState) {
-        return newUserState;
+    }
+  };
+
+  React.useEffect(() => {
+    setIsNavOpen(false);
+  }, [setIsNavOpen]);
+
+  React.useEffect(() => {
+    if (shouldRedirect) {
+      if (notebookLoaded && notebook) {
+        if (!isNotebookRunning) {
+          setStartShown(true);
+        } else {
+          history.push('/notebookController');
```

Comment:
```suggestion
          history.replace('/notebookController');
```
We are navigating our infra, we shouldn't put that into the history.

Reply:
Updated.